src/pudl/transform/ferc714.py

```diff
@@ -0,0 +1,483 @@
+        # Daily totals and null values
+        .query("hour25_mwh>=0.5*daily_mwh | hour25_mwh==0.0")
+        .loc[:, ["respondent_id", "plan_date"]]
+        .assign(hour=25)
+    )
+    idx_cols = ["respondent_id", "plan_date", "hour"]
+    fixed_df = (
+        fixed_df.set_index(idx_cols)
+        .drop(bad_totals.set_index(idx_cols).index)
+        .reset_index()
+        .set_index(["respondent_id", "plan_date"])
+    )
+
+    # Shift all the integer *after* the time change back by an hour, bu
+    # only do it for the days & respondents with real 25th hour data.
+    # Pandas can parse the duplicate hours appropriately in UTC conversion
+    hours_to_shift = (
+        fixed_df.loc[fixed_df.query("hour==25").index]
+        .query("hour>2")
+    )
+    fixed_df.loc[hours_to_shift.index,
+                 "hour"] = fixed_df.loc[hours_to_shift.index, "hour"] - 1
+    # Use the hour field to create a datetime field, switch to 0-23hr notation.
+    fixed_df = fixed_df.reset_index()
+    fixed_df["local_time"] = fixed_df.plan_date + \
+        pd.to_timedelta(fixed_df.hour - 1, unit="h")
+
+    # Find and replace the 0.0 values at 2-3am on the spring_forward days.
+    spring_forward_dates = list(pd.to_datetime(spring_forward).date)  # noqa: F841
+    demand_gaps = (
+        fixed_df
+        .query("plan_date in @spring_forward_dates")
+        .query("hour in [2,3]")
+        .query("demand_mwh==0.0")
+    )
+    fixed_df.loc[demand_gaps.index, "demand_mwh"] = np.nan
+    # Now that we have a real datetime column, ditch "hours" and "plan_date"
+    fixed_df = fixed_df.loc[:, ["respondent_id", "local_time",
+                                "timezone", "demand_mwh"]]
+
+    return fixed_df
+
+
+def _fix_tz_by_date(df, time_changes, respondent_id,
+                    start_time, end_time, tz_col, std_val, dst_val):
+    """Assign timezone abbreviations based on date."""
+    df = df.copy()
+    years = df.loc[(df.respondent_id == respondent_id) &
+                   (df.local_time >= start_time) &
+                   (df.local_time <= end_time)].local_time.dt.year.unique()
+    for yr in years:
+        spring = time_changes.loc[yr, "spring_forward"]
+        fall = time_changes.loc[yr, "fall_back"]
+        std_times = (
+            (df.respondent_id == respondent_id) &
+            (df.local_time.dt.year == yr) &
+            ((df.local_time < spring) | (df.local_time >= fall))
+        )
+        dst_times = (
+            (df.respondent_id == respondent_id) &
+            (df.local_time.dt.year == yr) &
+            ((df.local_time >= spring) & (df.local_time < fall))
+        )
+        df.loc[std_times, tz_col] = std_val
+        df.loc[dst_times, tz_col] = dst_val
+    return df
+
+
+def _apply_tz_fixes_by_date(df, fixes, time_changes):
+    """A wrapper assin many timezone fixes at once."""
+    for fix in fixes:
+        df = _fix_tz_by_date(df, time_changes=time_changes, **fix)
+    return df
+
+
```

Comment:
It seems following functions all basically generate one table each. Is this true? I know this is a WIP, but it would help me even just in the review process to document what purpose the purpose/desired outcome of these functions are.

Reply:
Yes, that's the idea -- that these are prototypes for the per-table transform functions that take as input the raw "page" building blocks and output the transformed dataframes, ready for output to data packages.